ui/lib/components/accordion/accordion.dart

```diff
@@ -109,10 +118,18 @@ class Accordion {
   }
 
   void removeItem(String id) {
-    _accordionItems.removeWhere((item) => item.id == id);
+    _accordionItems.removeWhere((item) => item._id == id);
     renderElement.children.removeWhere((item) => item.dataset['id'] == 'accordion-item-${id}');
   }
 
+  void updateItem(String id, AccordionItem item) {
+    _accordionItems.forEach((accordionItem) {
+      if (item._id == id) {
+        accordionItem = item;
```

Comment:
this won't work because `accordionItem` is a local variable, so changing its value won't replace it in the list.
I think the easiest way would be to find the index of the `accordionItem` with the same id, and then set the item at that index as the new item.

Another question - should `renderElement` also be updated with the new item, as we do for `appendItem` and `removeItem`?

Reply:
:O No idea why i thought this worked when i integrated with nook 🤦🏽 

For the renderElement, Since we construct the renderElement inside of the `item` that is passed into the updateItem - it seems to work. Unless I'm missing something..